### ISSUE TITLE
The loss curve of Dream-base

### ISSUE THREAD

**Issue by zhanzheng8585** [zhijie-group/Discrete-Diffusion-Forcing#7]
<img width="613" height="470" alt="Image" src="https://github.com/user-attachments/assets/9e8e821a-e79d-40b1-92d3-a1c9a9bc395f" />

Hi guys,

When training with the provided scripts, the raw per-step loss is highly volatile. After smoothing/averaging, the curve looks almost flat (see screenshot).

Are this variance and near-flat smoothed trend expected for D2F on this task, or am I misconfiguring something?

**Comment by xuchenkai1** [comment]
Thank you for your observation and detailed comparison.

Our loss curve is indeed generally similar to what you observed: there is an initial drop as the model quickly fits the block-wise generation pattern, and afterwards the loss does not decrease significantly, though performance continues to improve. This is expected behavior.

As for the large difference in loss magnitude between the Dream and llada models, the main reason is that llada uses an instruct model. For the padded regions after the input sequence, this type of model almost always predicts EOS. As a result, the KL divergence between the student and teacher predictions in those positions is very small, which lowers the average loss across all positions and leads to a smaller overall training loss.
